client/web/src/onboarding/LicenseModal.tsx

```diff
@@ -0,0 +1,179 @@
+import { type FC, useCallback, useEffect, useState, type FormEvent } from 'react'
+
+import type { ApolloQueryResult } from '@apollo/client'
+import classnames from 'classnames'
+import { format, formatDistanceToNow } from 'date-fns'
+import * as jsonc from 'jsonc-parser'
+import { useNavigate } from 'react-router-dom'
+import type { SiteConfigResult } from 'src/graphql-operations'
+import { useDebouncedCallback } from 'use-debounce'
+
+import { useIsLightTheme } from '@sourcegraph/shared/src/theme'
+import { Alert, Button, Form, H3, H2, Input, Label, LoadingSpinner, Modal, Text } from '@sourcegraph/wildcard'
+
+import { BrandLogo } from '../components/branding/BrandLogo'
+
+import { DEBOUNCE_DELAY_MS, DEFAULT_FORMAT_OPTIONS, DEFAULT_LICENSE_KEY_INFO, MIN_INPUT_LENGTH } from './constants'
+import { useUpdateLicenseKeyMutation } from './graphql-hooks'
+import type { LicenseInfo, LicenseKeyInfo } from './types'
+
+import styles from './LicenseModal.module.scss'
+
+interface LicenseKeyModalProps {
+    id: number
+    config: string
+    licenseKey: LicenseInfo
+    refetch?: () => Promise<ApolloQueryResult<SiteConfigResult>>
+    onHandleLicenseCheck: (
+        newValue: boolean | ((previousValue: boolean | undefined) => boolean | undefined) | undefined
+    ) => void
+}
+
+export const LicenseKeyModal: FC<LicenseKeyModalProps> = ({
+    licenseKey,
+    onHandleLicenseCheck,
+    config,
+    id,
+    refetch,
+}): JSX.Element => {
```

Comment:
The `FC<LicenseKeyModalProps>` type already provides the return type.

```suggestion
}) => {
```